enhancements/installer/aws-load-balancer-type.md

```diff
@@ -0,0 +1,313 @@
+
+It will look as follows -
+```go
+package aws
+
+import "github.com/aws/aws-sdk-go/aws/endpoints"
+
+// Platform stores all the global configuration that all machinesets
+// use.
+type Platform struct {
+
+// LBType allows user to set a load balancer type.
+// Allowed values are nlb or elb
+// +optional
+LBType string `json:"lbType,omitempty"`
+}
+```
+
+For the either infra config cluster object or ingress config cluster object we will need to add api for AwsLBType which will store and persist the information
+of the desired LB type mentioned by the installer config API so that the ingress operator will refer the LB type from this object whenever someone accidentally deletes the ingress controller.
+Currently, the default LB type i.e CLB/ELB gets set if you delete the ingress controller object even when the desired the LB type for default ingress controller
+was set to NLB as ingress operator does not have a way to check the LB type as we don't store it.
+
+If set it in the ingress object then the API will look as follows -
+Code location for Option 1 and Option 2 will be [here](https://github.com/openshift/api/blob/master/config/v1/types_ingress.go)
+
+Option 1:
+```go
+type IngressSpec struct {
+// awsLBType is used to set type of the LB either to CLB/ELB or NLB for the 
+// default ingress controller in AWS environment.
+// +optional
+AwsLBType string `json:"awsLBType,omitempty"`
+	
+```
+
+OR
+
+Option 2:
+```go
+type IngressSpec struct {
+// platformSpec holds desired information specific to the underlying
+// infrastructure provider related to ingress routing.
+PlatformSpec PlatformSpec `json:"platformSpec,omitempty"`
+
+// AWS contains settings specific to the Amazon Web Services infrastructure provider 
+// related to ingress routing.
+// +optional
+AWS *AWSPlatformSpec `json:"aws,omitempty"`
+}
+
+// AWSPlatformSpec holds the desired state of the Amazon Web Services infrastructure provider 
+// related to ingress routing.
+// This only includes fields that can be modified in the cluster.
+type AWSPlatformSpec struct {
+// LBType is used to set type of the LB either to CLB/ELB or NLB for the 
+// default ingress controller in AWS environment.
+// +optional
+LBType string `json:"lbType,omitempty"`
+}
+
+```
+
+OR
+
+Option 3:
+If we set it in the infra then the API will look as follows -
+Code will be [here](https://github.com/openshift/api/blob/63b58097a95d1b053cf038f56f89afc42524f598/config/v1/types_infrastructure.go#L351)
+```go
+// AWSPlatformSpec holds the desired state of the Amazon Web Services infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type AWSPlatformSpec struct {
+// LBType is used to set type of the LB either to CLB/ELB or NLB for the 
+// default ingress controller in AWS environment.
+// +optional
+LBType string `json:"lbType,omitempty"`
+}
+```
```

Comment:
@barbacbd @patrickdillon @jstuever The main question regarding the API is as to where shall we add the LB type. Shall it be the infra cluster object or ingress cluster object ? In this enhancement I have proposed 3 options so which one would the installer prefer ? cc @Miciah

Reply:
I will be using infra object as it seems to have all the cloud related env. So closing this loop as I need to start implementing an urgent epic.